test/hardhat/AqueductV1Auction.spec.ts

```diff
@@ -561,4 +561,144 @@ describe("AqueductV1Auction", () => {
             .to.emit(new ethers.Contract(token1.address, erc20Abi, owner), "Transfer") // send funds back to user
             .withArgs(auction.address, other.address, "4540540540540540540");
     });
+
+    /**************************************************************************
+     * tests below are related to exploits found in the first audit
+     *************************************************************************/
+
+    // issue: Reentrancy with a fake token in AqueductV1Auction | critical | (test #1)
+    // added reentrancy lock to placeBid(), test that the exploit contract now fails
+    it("auction:reentrancy-hack", async () => {
+        const { factory, wallet, token0, auction, attacker } = await loadFixture(fixture);
+
+        const exploitToken = await (await ethers.getContractFactory("ExploitToken")).connect(attacker).deploy();
+        await factory.createPair(token0.address, exploitToken.address);
+        const pair = (await ethers.getContractFactory("AqueductV1Pair")).attach(
+            await factory.getPair(token0.address, exploitToken.address)
+        );
+
+        await exploitToken.connect(attacker).initialize(auction.address, pair.address);
+
+        const token0Amount = expandTo18Decimals(1000);
+        const token1Amount = expandTo18Decimals(1000);
+        const AucToken0Amount = expandTo18Decimals(100);
+        const AucToken1Amount = expandTo18Decimals(100);
+
+        await exploitToken.connect(wallet).mint(expandTo18Decimals(10000));
+        await exploitToken.connect(attacker).mint(expandTo18Decimals(10));
+
+        // add liquidity
+        await token0
+            .transfer({
+                receiver: pair.address,
+                amount: token0Amount,
+            })
+            .exec(wallet);
+        await exploitToken.connect(wallet).transfer(pair.address, token1Amount);
+        await pair.mint(wallet.address);
+
+        await exploitToken.connect(wallet).transfer(attacker.address, expandTo18Decimals(10));
+
+        // Imagine the auction contract is active
+        await token0
+            .transfer({
+                receiver: auction.address,
+                amount: AucToken0Amount,
+            })
+            .exec(wallet);
+
+        await exploitToken.connect(wallet).transfer(auction.address, AucToken1Amount);
+
+        // exploit
+        const swapExploitAmount = expandTo18Decimals(10);
+        const bidExploitAmount = expandTo18Decimals(10000000);
+
+        // without this transfer swap in placeBid() will fail
+        await exploitToken.connect(attacker).transfer(pair.address, swapExploitAmount);
+
+        // transfer helper will fail with AUCTION_LOCKED, but reverts with its own error message
```

Comment:
Thoughts on a unit test to verify that `AUCTION_LOCKED` is indeed thrown under the right conditions in `AqueductV1Auction.sol`? Not sure how necessary it is tbh but let me know your thoughts